core/src/main/java/hudson/model/Executor.java

```diff
@@ -835,21 +836,39 @@ public void start() {
     @RequirePOST
     @Deprecated
     public void doStop( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
-        doStop().generateResponse(req,rsp,this);
+        doStop(req != null ? Util.fixEmpty(req.getParameter("what")) : null).generateResponse(req,rsp,this);
     }
 
     /**
      * Stops the current build.
      *
      * @since 1.489
+     * @deprecated as of 2.???
+     *      Use {@link #doStop(String)}.
      */
     @RequirePOST
+    @Deprecated
```

Comment:
Certainly wrong to deprecate it as-is actually, since that's what `AbstractBuild.doStop()` calls (as well as the `StopBuildsCommand.stopBuild()` CLI). My intent here was rather to document that it should no more be used as a stapler method, since the version with a query parameter would be preferred. Which doesn't mean it shouldn't be used from Java code anymore.  I don't know how to do that. Is there an annotation in stapler to ignore a `doSomething()` method?

Reply:
> Maybe I should simply add a new `doStopBuild(what)` method instead of messing with `doStop()` / `doStop(what)`.

Yep, saying it out loud, it sound better. Did that in 023098b082. Also renamed `what` as `runExtId` to be super explicit.